### DISCUSSION TITLE
iOS在cell中使用PAGImageView的setComposition方法，replaceImage替换图片，播放的pag会有闪烁的问题

### DISCUSSION THREAD

**Discussion by chenshengwen** [Tencent/libpag#2024] [Q&A] [answered]
在tableVIew的cell中使用一个PAGImageView，替换图片后第一次播放，就刚开始播放时会有闪烁问题，播放稳定后才不会闪烁

@property (nonatomic ,strong) PAGImageView *giftPagView;

- (void)playPagView:(UIImage *)image {
    
    NSString *filePath = [TUIProjectHelper getResourcePathForName:@"chat_receive_normal" extension:@"pag"];
    PAGFile *file = [PAGFile Load:filePath];
    self.giftPagView.hidden = NO;
    if (!self.giftPagView.isPlaying) {
        PAGImage *pagImage = [PAGImage FromCGImage:[image CGImage]];
        [file replaceImage:1 data:pagImage];
        [self.giftPagView setComposition:file];
        [self.giftPagView play];
    }
}

如果把  [file replaceImage:1 data:pagImage];这里注释掉后，就没有闪烁问题，然后我试了在UIView中替换图片就没有这种问题，同一个pag文件在安卓端播放没有问题

**Comment by kevingpqi123** [comment, marked as answer]
如果涉及到全屏播放、占位图替换和文本编辑中的一种，不推荐使用 PAGImageView，此时渲染数据不会被持久化，每次启动还是要渲染，这是推荐使用 PAGView

**Reply by chenshengwen** [reply]
使用PAGView后，在tableView中确实不会闪烁，但是在tableView不断的上下滑动带有PAGView的cell，会崩溃，而目前我在cell中的PAGImageView大小是200pt*200pt

**Reply by kevingpqi123** [reply]
那就基于 pag 的 demo 工程看下能否构造出崩溃的场景，提供复现的 demo及复现步骤
